### PR TITLE
Use of enum types in Peep code

### PR DESCRIPTION
**Pull request by richard-fine**
The big rct_peep struct has a number of fields which hold enumerated values, but they have just been typed as uint8_t, which means that we get no type checking by the compiler and that the debugger does not visualize values in helpful ways.

This PR changes the relevant fields, and some code that operates on them, to use the enum type directly. In order to ensure that this does not change the field width, we use C++11's support for explicitly declaring the width of an enum.

In the process, I also went and figured out the names for a couple of untitled action animation states (PEEP_ACTION_SPRITE_TYPE_2 and PEEP_ACTION_SPRITE_TYPE_7) by looking at the G1 sprite graphics and at the places where the state is used.

### CODE REVIEW

File: src/openrct2/peep/Peep.h [hunk 1]
```diff
@@ -249,7 +251,7 @@ enum PEEP_USING_BIN_SUB_STATE
     PEEP_USING_BIN_GOING_BACK,
 };
 
-enum PEEP_ACTION_EVENTS
+enum PEEP_ACTION_EVENTS : uint8_t
```
Comment (ZehMatt) [inline src/openrct2/peep/Peep.h]: I feel like the name is slightly misleading, they aren't exactly events and the type can only hold one value at a time. Maybe we could rename it to something like PEEP_ACTION_TYPE for consistency.

**Comment by AaronVanGeffen** [comment]
Nice work — changing them into proper types is something I've been meaning to look into.

I do think we should change the type names to use UpperCamelCase (e.g. PEEP_ACTION_TYPE to PeepActionType), which is what we've been using in newer code.

**Comment by richard-fine** [comment]
OK, I'll do some renames.

In the long run we also have the option of making them enum classes rather than just enums - that would be a much bigger-impact change as it would mean all use of enum values would have to become scoped. Not for this PR but something to consider in the future.

File: src/openrct2/peep/Peep.h (hunk 1 above)
Comment (richard-fine) [reply src/openrct2/peep/Peep.h]: Renamed to PeepActionType in 60396c665861f8541b8d12a88414b538e209cb2f.

**Comment by richard-fine** [comment]
I think this is ready to go now.

**Pull request merged**